### PR TITLE
Start shipyard phase for v0.9.0-m2

### PR DESCRIPTION
**Pull request by mangelajo**
This also updates the submariner image to be submariner-gateway since
this changed in m2.

Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

File: releases/v0.9.0-m2.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+version: v0.9.0-m2
+name: 0.9.0-m2
+pre-release: true
+release-notes: https://github.com/submariner-io/submariner-website/pull/457
+status: shipyard
+components:
+  shipyard: 6998e339fcab21814d70a07c276d43ddf269b1e6
+
```
Comment (skitt) [inline releases/v0.9.0-m2.yaml]: The blank line here trips up `yamllint`.

**Review by skitt — CHANGES_REQUESTED** [review]

**Comment by skitt** [comment]
Oh, apparently dropping the git-tagged images has broken the release e2e...

**Comment by mangelajo** [comment]
> Oh, apparently dropping the git-tagged images has broken the release e2e...

oh did we drop git-tagged images? :?

**Comment by mangelajo** [comment]
> Oh, apparently dropping the git-tagged images has broken the release e2e...

we need them back

File: releases/v0.9.0-m2.yaml (hunk 1 above)
Comment (mangelajo) [reply releases/v0.9.0-m2.yaml]: Done

**Comment by mangelajo** [comment]
> > Oh, apparently dropping the git-tagged images has broken the release e2e...
> 
> we need them back

We actually had them, just the image name had changed :)

**Review by skitt — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Comment by mkolesnik** [comment]
Seems theres some trouble building the subctl binary, some sneaky `w` is sneaking in somehow..

From the E2E log:
```
[submariner-operator]$ make --debug=b bin/subctl w --debug=b -- using=
```

This happens inside the dapper, but outside the call is simply `make bin/subctl`

**Comment by mangelajo** [comment]
E2E needs rerun after the new shipyard images generate. https://github.com/submariner-io/shipyard/runs/2238861833

**Comment by tpantelis** [comment]
E2E is failing.  @mkolesnik can you look into this please?

**Comment by mkolesnik** [comment]
> E2E is failing. @mkolesnik can you look into this please?

So there's been a regression https://github.com/submariner-io/submariner-operator/issues/1209
I've created #109 which should fix it here

**Comment by mkolesnik** [comment]
Seems we also need #110 though I'm not sure why it didnt interfere before

**Pull request merged**